rpi/page.py

```diff
@@ -0,0 +1,336 @@
+        
+		# Show failure
+		self.title_label.config(text="Error authenticating user!", font="helvetica 20 bold")
+		self.title_label.pack(side="top", fill="both", expand=True)
+
+	def cast_vote(self, candidate):
+		# TO-DO: logic for vote casting here
+		self.fingerprint_auth_result_string.set(f"Vote for Candidate {candidate} cast successfully!")
+		self.farewell_message()
+
+	def farewell_message(self):
+		# Clear existing components
+		self.title_label.pack_forget()
+		self.fingerprint_auth_result_message.pack_forget()
+		self.instruction_label.pack_forget()
+		self.vote_button_1.pack_forget()
+		self.vote_button_2.pack_forget()
+		self.vote_button_3.pack_forget()
+		# Show farewell msg
+		self.title_label.config(text="Thank you for voting!", font="helvetica 20 bold")
+		self.title_label.pack(side="top", fill="both", expand=True)
+
+        
+class LoginPage(Page):
+	def __init__(self, main_view, *args, **kwargs):
+		Page.__init__(self, *args, **kwargs)
+		self.main_view = main_view  # Store a reference to the MainView instance
+		self.title_label = tk.Label(self, text="Login to system", font="helvetica 20 bold", bg="#fff")
+		self.title_label.pack(side="top", fill="both", expand=True)
+		self.username_label = tk.Label(self, bg="#fff", text="Username")
+		self.username_input = tk.Entry(self, width=20, bg="#fff", bd=1, relief="solid")
+		self.password_label = tk.Label(self, bg="#fff", text="Password")
+		self.password_input = tk.Entry(self, show="*", width=20, bg="#fff", bd=1, relief="solid")
+		self.submit_credentials_btn = tk.Button(self, text="Submit", height=1, width=8, command=lambda: attempt_login(self, self.username_input.get(), self.password_input.get()))
+		self.register_user_btn = tk.Button(self, text="Don't have an account? Click here to register.", bg="#fff", font="helvetica 12 underline", command=lambda: self.main_view.show_register_page())
+		self.result_string = tk.StringVar(value="")
+		self.result_message = tk.Label(self, textvariable=self.result_string, wraplength=500, justify="center", bg="#fff", font="helvetica 14")
+		self.username_label.pack(pady=10)
+		self.username_input.pack(pady=5)
+		self.password_label.pack(pady=10)
+		self.password_input.pack(pady=5)
+		self.submit_credentials_btn.pack(pady=30)
+		self.register_user_btn.pack(pady=10)
+		
+		def attempt_login(self, username, password):
+			credentials = {"username": username, "password": password}
+			server_response = None
+			try:
+				server_response = requests.post("https://fingerprint-voter-server.onrender.com/login", data=credentials)
+				if server_response.status_code == requests.codes.ok:
+					alert.show_alert(alert.AlertType.SUCCESS, "Logged in successfully", 2.5, self.result_string, self.result_message)
+				
+				# Get the JWT token from the server
+					jwt_token = server_response.json().get("token")
+
+				# Store the JWT token
+					kr.set_password("fp-voter", "jwt_token", jwt_token)
+					
+					self.main_view.show_vote_page()
```

Comment:
Server now does return a field called **fingerprintId** IIRC. Should be used to search fingerprint as well.